standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -1248,30 +1253,43 @@ protected boolean shouldStop() {
     }
 
     private static enum FilterType {
-      Integral,
-      String,
-      Date,
-
-      Invalid;
-
-      static FilterType fromType(String colTypeStr) {
-        if (colTypeStr.equals(ColumnType.STRING_TYPE_NAME)) {
-          return FilterType.String;
-        } else if (colTypeStr.equals(ColumnType.DATE_TYPE_NAME)) {
-          return FilterType.Date;
-        } else if (ColumnType.IntegralTypes.contains(colTypeStr)) {
-          return FilterType.Integral;
+      Integral("tinyint,smallint,int,bigint", "java.lang.Long"),
+      String("string", "java.lang.String"),
+      Date("date", "java.sql.Date"),
+      Timestamp("timestamp", "java.sql.Timestamp"),
+
+      Invalid("Invalid", "null");
+
+      private final List<String> type;
+      private final String value;
+
+      FilterType(String type, String value) {
+        this.type = Arrays.asList(type.split(","));
+        this.value = value;
+      }
+
+      public List<String> getType() {
+        return type;
+      }
+
+      public Object getValue() {
+        return value;
+      }
+
+      public static FilterType fromType(String type) {
+        for (FilterType filterType : FilterType.values()) {
+          if (filterType.type.stream().anyMatch(val -> val.equalsIgnoreCase(type))) {
```

Comment:
By using `Set` this check can be simplified to
```
if (filterType.colTypes.contains(colTypeStr)) {
```

Reply:
ack